### PR TITLE
feat: Add addresses first pass (SC-365)

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

File: src/Ethereum.sol [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    /******************************************************************************************************************/
+    /*** Token Addresses                                                                                            ***/
+    /******************************************************************************************************************/
+
+    address internal constant DAI    = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address internal constant SDAI   = 0x83F20F44975D03b1b09e64809B757c47f942BEeA;
+    address internal constant USDC   = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address internal constant WBTC   = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+    address internal constant WETH   = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address internal constant WSTETH = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
```
Comment (hexonaut) [inline src/Ethereum.sol:30]: Add RETH, GNO and USDT.

File: src/Ethereum.sol [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    /******************************************************************************************************************/
+    /*** Core Protocol Addresses                                                                                    ***/
+    /******************************************************************************************************************/
+
+    address internal constant AAVE_ORACLE                      = 0x8105f69D9C41644c6A0803fDA7D03Aa70996cFD9;
+    address internal constant ACL_MANAGER                      = 0xdA135Cd78A086025BcdC87B038a1C462032b510C;
+    address internal constant DAI_TREASURY                     = 0x856900aa78e856a5df1a2665eE3a66b2487cD68f;  // Deprecated?
```
Comment (hexonaut) [inline src/Ethereum.sol:38]: It won't be used, but it's still the treasury for DAI. I would remove the `deprecated` note.

File: src/Ethereum.sol [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant TREASURY_CONTROLLER              = 0x92eF091C5a1E01b3CE1ba0D0150C84412d818F7a;
+    address internal constant WETH_GATEWAY                     = 0xBD7D6a9ad7865463DE44B05F04559f65e3B11704;
+
+    /******************************************************************************************************************/
+    /*** Reserve Addresses                                                                                          ***/
+    /******************************************************************************************************************/
+
+    address internal constant DAI_ATOKEN            = 0x4DEDf26112B3Ec8eC46e7E31EA5e123490B05B8B;
+    address internal constant DAI_IRM               = 0x7949a8Ef09c49506cCB1cB983317272dcf4170Dd;
+    address internal constant DAI_ORACLE            = 0xAed0c38402a5d19df6E4c03F4E2DceD6e29c1ee9;
```
Comment (hexonaut) [inline src/Ethereum.sol:55]: I would remove the IRM and Oracle. They change frequently enough to likely become out of date here.

File: src/Ethereum.sol [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant DAI_STABLE_DEBT_TOKEN = 0xfe2B7a7F4cC0Fb76f7Fc1C6518D586F1e4559176;
+    address internal constant DAI_DEBT_TOKEN        = 0xf705d2B7e92B3F38e6ae7afaDAA2fEE110fE5914;
+
+    address internal constant SDAI_ATOKEN            = 0x78f897F0fE2d3B5690EbAe7f19862DEacedF10a7;
+    address internal constant SDAI_IRM               = 0xeC4cf692c18E62159a39704Aa1Db82ca2306fF90;
+    address internal constant SDAI_ORACLE            = 0xb9E6DBFa4De19CCed908BcbFe1d015190678AB5f;
```
Comment (hexonaut) [inline src/Ethereum.sol:61]: Same here.

File: src/Ethereum.sol [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant SDAI_STABLE_DEBT_TOKEN = 0xEc6C6aBEd4DC03299EFf82Ac8A0A83643d3cB335;
+    address internal constant SDAI_DEBT_TOKEN        = 0xaBc57081C04D921388240393ec4088Aa47c6832B;
+
+    address internal constant USDC_ATOKEN            = 0x377C3bd93f2a2984E1E7bE6A5C22c525eD4A4815;
+    address internal constant USDC_IRM               = 0x0F1a9a787b4103eF5929121CD9399224c6455dD6;
+    address internal constant USDC_ORACLE            = 0x98e6BcBA7d5daFbfa4a92dAF08d3d7512820c30C;
```
Comment (hexonaut) [inline src/Ethereum.sol:67]: Same here.

File: src/Ethereum.sol [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant USDC_STABLE_DEBT_TOKEN = 0x887Ac022983Ff083AEb623923789052A955C6798;
+    address internal constant USDC_DEBT_TOKEN        = 0x7B70D04099CB9cfb1Db7B6820baDAfB4C5C70A67;
+
+    address internal constant WBTC_ATOKEN            = 0x4197ba364AE6698015AE5c1468f54087602715b2;
+    address internal constant WBTC_IRM               = 0xD2139d6d63Acb1e7Cc91cE32bbD86eFb17eBEe46;
+    address internal constant WBTC_ORACLE            = 0x230E0321Cf38F09e247e50Afc7801EA2351fe56F;
```
Comment (hexonaut) [inline src/Ethereum.sol:73]: Same here.

File: src/Ethereum.sol [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant WBTC_STABLE_DEBT_TOKEN = 0x4b29e6cBeE62935CfC92efcB3839eD2c2F35C1d9;
+    address internal constant WBTC_DEBT_TOKEN        = 0xf6fEe3A8aC8040C3d6d81d9A4a168516Ec9B51D2;
+
+    address internal constant WETH_ATOKEN            = 0x59cD1C87501baa753d0B5B5Ab5D8416A45cD71DB;
+    address internal constant WETH_IRM               = 0xD7A8461e6aF708a086D8285f8fD900309336347c;
+    address internal constant WETH_ORACLE            = 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419;
```
Comment (hexonaut) [inline src/Ethereum.sol:79]: Same here.

File: src/Ethereum.sol [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant WETH_STABLE_DEBT_TOKEN = 0x3c6b93D38ffA15ea995D1BC950d5D0Fa6b22bD05;
+    address internal constant WETH_DEBT_TOKEN        = 0x2e7576042566f8D6990e07A1B61Ad1efd86Ae70d;
+
+    address internal constant WSTETH_ATOKEN            = 0x12B54025C112Aa61fAce2CDB7118740875A566E9;
+    address internal constant WSTETH_IRM               = 0x0D56700c90a690D8795D6C148aCD94b12932f4E3;
+    address internal constant WSTETH_ORACLE            = 0xA9F30e6ED4098e9439B2ac8aEA2d3fc26BcEbb45;
```
Comment (hexonaut) [inline src/Ethereum.sol:85]: Same here.

File: src/Ethereum.sol [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant WSTETH_STABLE_DEBT_TOKEN = 0x9832D969a0c8662D98fFf334A4ba7FeE62b109C2;
+    address internal constant WSTETH_DEBT_TOKEN        = 0xd5c3E3B566a42A6110513Ac7670C1a86D76E13E6;
```
Comment (hexonaut) [inline src/Ethereum.sol:87]: Add token addresses for GNO, RETH and USDT.

File: src/Ethereum.sol [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant UI_INCENTIVE_DATA_PROVIDER = 0xA7F8A757C4f7696c015B595F51B2901AC0121B18;
+    address internal constant UI_POOL_DATA_PROVIDER      = 0xF028c2F4b19898718fD0F77b9b881CbfdAa5e8Bb;
+    address internal constant WALLET_BALANCE_PROVIDER    = 0xd2AeF86F51F92E8e49F42454c287AE4879D1BeDc;
+
+    /******************************************************************************************************************/
+    /*** Implementation Addresses                                                                                   ***/
+    /******************************************************************************************************************/
+
+    address internal constant A_TOKEN_IMPL             = 0x6175ddEc3B9b38c88157C10A01ed4A3fa8639cC6;
+    address internal constant DAI_TREASURY_IMPL        = 0xF1E57711Eb5F897b415de1aEFCB64d9BAe58D312;  // Deprecated?
```
Comment (hexonaut) [inline src/Ethereum.sol:103]: Not deprecated. Include it.

File: src/Ethereum.sol [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+    address internal constant INCENTIVES_IMPL          = 0x0ee554F6A1f7a4Cb4f82D4C124DdC2AD3E37fde1;
+    address internal constant POOL_CONFIGURATOR_IMPL   = 0xF7b656C95420194b79687fc86D965FB51DA4799F;
+    address internal constant POOL_IMPL                = 0x8115366Ca7Cf280a760f0bC0F6Db3026e2437115;
```
Comment (hexonaut) [inline src/Ethereum.sol:106]: We updated this to `0xb40f6d584081ac2b0fd84c846dba3c1417889304`.

File: src/Gnosis.sol [hunk 12]
```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.13;
+
+library Gnosis {
+
```
Comment (hexonaut) [inline src/Gnosis.sol:5]: I would include the AMB (L1 and L2) and L2 Executor (admin). I would also include the tokens added to the market.

File: src/Gnosis.sol [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    /******************************************************************************************************************/
+    /*** Core Protocol Addresses                                                                                    ***/
+    /******************************************************************************************************************/
+
+    address internal constant AAVE_ORACLE                      = 0x8105f69D9C41644c6A0803fDA7D03Aa70996cFD9;
+    address internal constant ACL_MANAGER                      = 0x86C71796CcDB31c3997F8Ec5C2E3dB3e9e40b985;
+    address internal constant DAI_TREASURY                     = 0x856900aa78e856a5df1a2665eE3a66b2487cD68f;  // Deprecated?
```
Comment (hexonaut) [inline src/Gnosis.sol:12]: There is no dai treasury on Gnosis.

File: src/Gnosis.sol [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    address internal constant EMISSION_MANAGER                 = 0xf09e48dd4CA8e76F63a57ADd428bB06fee7932a4;
```
Comment (hexonaut) [inline src/Gnosis.sol:13]: This address is wrong.

File: src/Gnosis.sol [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    address internal constant UI_INCENTIVE_DATA_PROVIDER = 0xA7F8A757C4f7696c015B595F51B2901AC0121B18;
+    address internal constant UI_POOL_DATA_PROVIDER      = 0xF028c2F4b19898718fD0F77b9b881CbfdAa5e8Bb;
+    address internal constant WALLET_BALANCE_PROVIDER    = 0xd2AeF86F51F92E8e49F42454c287AE4879D1BeDc;
+
+    /******************************************************************************************************************/
+    /*** Implementation Addresses                                                                                   ***/
+    /******************************************************************************************************************/
+
+    address internal constant A_TOKEN_IMPL             = 0x856900aa78e856a5df1a2665eE3a66b2487cD68f;
+    address internal constant DAI_TREASURY_IMPL        = 0xF1E57711Eb5F897b415de1aEFCB64d9BAe58D312;  // Deprecated?
```
Comment (hexonaut) [inline src/Gnosis.sol:37]: There is no dai treasury on Gnosis.

File: src/Gnosis.sol [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    address internal constant INCENTIVES_IMPL          = 0x764b4AB9bCA18eB633d92368F725765Ebb8f047C;
+    address internal constant POOL_CONFIGURATOR_IMPL   = 0x6175ddEc3B9b38c88157C10A01ed4A3fa8639cC6;
+    address internal constant POOL_IMPL                = 0x026a5B6114431d8F3eF2fA0E1B2EDdDccA9c540E;
```
Comment (hexonaut) [inline src/Gnosis.sol:40]: This is out of date. We updated to `0xa8fC41696F2a230b03F77d258Db39069e9e55F56`.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Good start. Some recommendations.

File: src/Ethereum.sol (hunk 1 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:30]: Updated

File: src/Ethereum.sol (hunk 2 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:38]: Removed

File: src/Ethereum.sol (hunk 3 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:55]: Removed

File: src/Ethereum.sol (hunk 4 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:61]: Removed

File: src/Ethereum.sol (hunk 5 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:67]: Removed

File: src/Ethereum.sol (hunk 6 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:73]: Removed

File: src/Ethereum.sol (hunk 7 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:79]: Removed

File: src/Ethereum.sol (hunk 8 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:85]: Removed

File: src/Ethereum.sol (hunk 10 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:103]: Removed comment

File: src/Ethereum.sol (hunk 11 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:106]: Updated

File: src/Gnosis.sol (hunk 13 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:12]: Removed

File: src/Gnosis.sol (hunk 15 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:37]: Removed

File: src/Gnosis.sol (hunk 16 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:40]: Updated

File: src/Gnosis.sol (hunk 12 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:5]: Added AMBs and executor

File: src/Ethereum.sol (hunk 9 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:90]: Added

File: src/Gnosis.sol (hunk 12 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:5]: Added AMBs, tokens, and reserve tokens

File: src/Gnosis.sol (hunk 14 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:12]: Fixed

File: src/Ethereum.sol [hunk 17]
```diff
@@ -47,65 +36,83 @@ library Ethereum {
...
 
+    address internal constant DAI                   = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
     address internal constant DAI_ATOKEN            = 0x4DEDf26112B3Ec8eC46e7E31EA5e123490B05B8B;
-    address internal constant DAI_IRM               = 0x7949a8Ef09c49506cCB1cB983317272dcf4170Dd;
-    address internal constant DAI_ORACLE            = 0xAed0c38402a5d19df6E4c03F4E2DceD6e29c1ee9;
     address internal constant DAI_STABLE_DEBT_TOKEN = 0xfe2B7a7F4cC0Fb76f7Fc1C6518D586F1e4559176;
     address internal constant DAI_DEBT_TOKEN        = 0xf705d2B7e92B3F38e6ae7afaDAA2fEE110fE5914;
 
+    address internal constant GNO                   = 0x6810e776880C02933D47DB1b9fc05908e5386b96;
+    address internal constant GNO_ATOKEN            = 0xbf13910620722D4D4F8A03962894EB3335Bf4FaE;
```
Comment (hexonaut) [inline src/Ethereum.sol:48]: I think you're pulling the wrong addresses from the pool `getReserveData` - off by one.

This one is `0x7b481aCC9fDADDc9af2cBEA1Ff2342CB1733E50F`.

File: src/Ethereum.sol [hunk 18]
```diff
@@ -47,65 +36,83 @@ library Ethereum {
...
+    address internal constant GNO_STABLE_DEBT_TOKEN = 0x57a2957651DA467fCD4104D749f2F3684784c25a;
```
Comment (hexonaut) [inline src/Ethereum.sol:49]: This is `0xbf13910620722D4D4F8A03962894EB3335Bf4FaE`.

File: src/Ethereum.sol [hunk 19]
```diff
@@ -47,65 +36,83 @@ library Ethereum {
...
+    address internal constant GNO_DEBT_TOKEN        = 0x554265A713D6746A62d86A797254590784D436AA;
```
Comment (hexonaut) [inline src/Ethereum.sol:50]: This one is `0x57a2957651DA467fCD4104D749f2F3684784c25a`.

File: src/Ethereum.sol [hunk 20]
```diff
@@ -47,65 +36,83 @@ library Ethereum {
...
     address internal constant USDC_ATOKEN            = 0x377C3bd93f2a2984E1E7bE6A5C22c525eD4A4815;
-    address internal constant USDC_IRM               = 0x0F1a9a787b4103eF5929121CD9399224c6455dD6;
-    address internal constant USDC_ORACLE            = 0x98e6BcBA7d5daFbfa4a92dAF08d3d7512820c30C;
     address internal constant USDC_STABLE_DEBT_TOKEN = 0x887Ac022983Ff083AEb623923789052A955C6798;
     address internal constant USDC_DEBT_TOKEN        = 0x7B70D04099CB9cfb1Db7B6820baDAfB4C5C70A67;
 
+    address internal constant USDT                   = 0xdAC17F958D2ee523a2206206994597C13D831ec7;
+    address internal constant USDT_ATOKEN            = 0x0Dae62F953Ceb2E969fB4dE85f3F9074fa920776;
+    address internal constant USDT_STABLE_DEBT_TOKEN = 0x529b6158d1D2992E3129F7C69E81a7c677dc3B12;
+    address internal constant USDT_DEBT_TOKEN        = 0x0F1a9a787b4103eF5929121CD9399224c6455dD6;
```
Comment (hexonaut) [inline src/Ethereum.sol:70]: These are in order: `0xe7dF13b8e3d6740fe17CBE928C7334243d86c92f`, `0x0Dae62F953Ceb2E969fB4dE85f3F9074fa920776` and `0x529b6158d1D2992E3129F7C69E81a7c677dc3B12`.

File: src/Gnosis.sol [hunk 21]
```diff
@@ -21,25 +20,65 @@ library Gnosis {
...
     /******************************************************************************************************************/
 
-    address internal constant PROTOCOL_DATA_PROVIDER     = 0x2a002054A06546bB5a264D57A81347e23Af91D18;
-    address internal constant UI_INCENTIVE_DATA_PROVIDER = 0xA7F8A757C4f7696c015B595F51B2901AC0121B18;
-    address internal constant UI_POOL_DATA_PROVIDER      = 0xF028c2F4b19898718fD0F77b9b881CbfdAa5e8Bb;
-    address internal constant WALLET_BALANCE_PROVIDER    = 0xd2AeF86F51F92E8e49F42454c287AE4879D1BeDc;
+    address internal constant GNO                   = 0x9C58BAcC331c9aa871AFD802DB6379a98e80CEdb;
+    address internal constant GNO_ATOKEN            = 0x2f589BADbE2024a94f144ef24344aF91dE21a33c;
+    address internal constant GNO_STABLE_DEBT_TOKEN = 0xd4bAbF714964E399f95A7bb94B3DeaF22d9F575d;
+    address internal constant GNO_DEBT_TOKEN        = 0xE74569197bFba0e2231c84587cc5f9ceDB807043;
```
Comment (hexonaut) [inline src/Gnosis.sol:29]: These are off by one again. Addresses are: `0x5671b0B8aC13DC7813D36B99C21c53F6cd376a14`, `0x2f589BADbE2024a94f144ef24344aF91dE21a33c`  and `0xd4bAbF714964E399f95A7bb94B3DeaF22d9F575d`.

File: src/Gnosis.sol [hunk 22]
```diff
@@ -21,25 +20,65 @@ library Gnosis {
...
+
+    address internal constant WETH                   = 0x6A023CCd1ff6F2045C3309768eAd9E68F978f6e1;
+    address internal constant WETH_ATOKEN            = 0xe21Bf3FB5A2b5Bf7BAE8c6F1696c4B097F5D2f93;
+    address internal constant WETH_STABLE_DEBT_TOKEN = 0x0aD6cCf9a2e81d4d48aB7db791e9da492967eb84;
+    address internal constant WETH_DEBT_TOKEN        = 0xC2C2F532334D5868508f65ddB4610602C6ee8c5d;
```
Comment (hexonaut) [inline src/Gnosis.sol:34]: Off by one: `0x629D562E92fED431122e865Cc650Bc6bdE6B96b0`,`0xe21Bf3FB5A2b5Bf7BAE8c6F1696c4B097F5D2f93`,`0x0aD6cCf9a2e81d4d48aB7db791e9da492967eb84`

File: src/Gnosis.sol [hunk 23]
```diff
@@ -21,25 +20,65 @@ library Gnosis {
...
+
+    address internal constant WSTETH                   = 0x6C76971f98945AE98dD7d4DFcA8711ebea946eA6;
+    address internal constant WSTETH_ATOKEN            = 0x0F0e336Ab69D9516A9acF448bC59eA0CE79E4a42;
+    address internal constant WSTETH_STABLE_DEBT_TOKEN = 0x3294dA2E28b29D1c08D556e2B86879d221256d31;
+    address internal constant WSTETH_DEBT_TOKEN        = 0x27c74895F0f49177cc4450Bf017281E556751a1a;
```
Comment (hexonaut) [inline src/Gnosis.sol:39]: Off by one: `0x9Ee4271E17E3a427678344fd2eE64663Cb78B4be`,`0x0F0e336Ab69D9516A9acF448bC59eA0CE79E4a42`,`0x3294dA2E28b29D1c08D556e2B86879d221256d31`

File: src/Gnosis.sol [hunk 24]
```diff
@@ -21,25 +20,65 @@ library Gnosis {
...
+
+    address internal constant WXDAI                   = 0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d;
+    address internal constant WXDAI_ATOKEN            = 0xab1B62A1346Acf534b581684940E2FD781F2EA22;
+    address internal constant WXDAI_STABLE_DEBT_TOKEN = 0x868ADfDf12A86422524EaB6978beAE08A0008F37;
+    address internal constant WXDAI_DEBT_TOKEN        = 0x8b0271080D53f759eCc607f13316bDB3B86449e2;
```
Comment (hexonaut) [inline src/Gnosis.sol:44]: Off by one: `0xC9Fe2D32E96Bb364c7d29f3663ed3b27E30767bB`,`0xab1B62A1346Acf534b581684940E2FD781F2EA22`,`0x868ADfDf12A86422524EaB6978beAE08A0008F37`

**Review by hexonaut — CHANGES_REQUESTED** [review]

File: src/Ethereum.sol (hunk 17 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:48]: Yep you're right - fixed

File: src/Ethereum.sol (hunk 18 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:49]: fixed

File: src/Gnosis.sol (hunk 21 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:29]: fixed

File: src/Gnosis.sol (hunk 23 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:39]: fixed

File: src/Ethereum.sol (hunk 20 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:70]: Fixed

File: src/Gnosis.sol (hunk 22 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:34]: Fixed

File: src/Gnosis.sol (hunk 24 above)
Comment (lucas-manuel) [reply src/Gnosis.sol:44]: Fixed

**Review by hexonaut — CHANGES_REQUESTED** [review]
Recent changes look good. Let's add the cap automator, freezer mom and Morpho vault addresses.

File: src/Ethereum.sol [hunk 25]
```diff
@@ -85,7 +86,18 @@ library Ethereum {
     address internal constant WSTETH_DEBT_TOKEN        = 0xd5c3E3B566a42A6110513Ac7670C1a86D76E13E6;
 
     /******************************************************************************************************************/
-    /*** Implementation Addresses                                                                                   ***/
+    /*** SparkLend - Auxiliary Protocol Addresses                                                                   ***/
```
Comment (hexonaut) [inline src/Ethereum.sol:89]: I would also add the kill switch oracle: `0x909A86f78e1cdEd68F9c2Fe2c9CD922c401abe82`

File: src/Ethereum.sol [hunk 26]
```diff
@@ -85,7 +86,18 @@ library Ethereum {
...
+    /******************************************************************************************************************/
+
+    address internal constant CAP_AUTOMATOR    = 0x2276f52afba7Cf2525fd0a050DF464AC8532d0ef;
+    address internal constant FREEZER_MOM      = 0xFA36c12Bc307b40c701D65d8FE8F88cCEdE2277a;
+    address internal constant SPELL_FREEZE_ALL = 0xA67d62f75F8D11395eE120CA8390Ab3bF01f0b8A;
+    address internal constant SPELL_FREEZE_DAI = 0x0F9149c4d6018A5999AdA5b592E372845cfeC725;
+    address internal constant SPELL_PAUSE_ALL  = 0x216738c7B1E83cC1A1FFcD3433226B0a3B174484;
+    address internal constant SPELL_PAUSE_DAI  = 0x1B94E2F3818E1D657bE2A62D37560514b52DB17F;
```
Comment (hexonaut) [inline src/Ethereum.sol:97]: This should be moved to a separate section for well-known emergency spells.

File: src/Ethereum.sol [hunk 27]
```diff
@@ -16,10 +16,11 @@ library Ethereum {
     /*** SparkDAO Addresses                                                                                         ***/
     /******************************************************************************************************************/
 
-    address internal constant SPARK_PROXY = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
+    address internal constant META_MORPHO_VAULT = 0x73e65DBD630f90604062f6E02fAb9138e713edD9;
+    address internal constant SPARK_PROXY      = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
```
Comment (hexonaut) [inline src/Ethereum.sol:20]: Alignment.

File: src/Ethereum.sol [hunk 28]
```diff
@@ -16,10 +16,11 @@ library Ethereum {
     /*** SparkDAO Addresses                                                                                         ***/
     /******************************************************************************************************************/
 
-    address internal constant SPARK_PROXY = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
+    address internal constant META_MORPHO_VAULT = 0x73e65DBD630f90604062f6E02fAb9138e713edD9;
```
Comment (hexonaut) [inline src/Ethereum.sol:19]: I would label this as `MORPHO_VAULT_DAI_1`.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Some more changes.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Let's also add more from Morpho. Add sUSDe and USDe tokens into a Morpho tokens section.

File: src/Ethereum.sol (hunk 27 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:20]: Fixed

File: src/Ethereum.sol (hunk 28 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:19]: Updated

File: src/Ethereum.sol (hunk 25 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:89]: Added

File: src/Ethereum.sol (hunk 26 above)
Comment (lucas-manuel) [reply src/Ethereum.sol:97]: Moved

File: src/Ethereum.sol [hunk 29]
```diff
@@ -16,8 +32,16 @@ library Ethereum {
...
+    address internal constant SPARK_PROXY = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
+
+    /******************************************************************************************************************/
+    /*** Morpho Addresses                                                                                           ***/
+    /******************************************************************************************************************/
+
+    address internal constant MORPHO_DEFAULT_IRM      = 0x870aC11D48B15DB9a138Cf899d20F13F79Ba00BC;
+    address internal constant MORPHO_STETH_ETH_ORACLE = 0x86392dC19c0b719886221c78AB11eb8Cf5c52812;
+    address internal constant MORPHO_VAULT_DAI_1      = 0x73e65DBD630f90604062f6E02fAb9138e713edD9;
+    address internal constant MORPHO_WBTC_BTC_ORACLE  = 0xfdFD9C85aD200c506Cf9e21F1FD8dd01932FBB23;
```
Comment (hexonaut) [inline src/Ethereum.sol:44]: These are Chainlink oracles and are for the kill switch. If you want to add the sUSDe and USDe oracles, they are `0x5D916980D5Ae1737a8330Bf24dF812b2911Aae25` and `0xaE4750d0813B5E37A51f7629beedd72AF1f9cA35` respectively.

**Review by hexonaut — CHANGES_REQUESTED** [review]
One more change.

**Review by hexonaut — APPROVED** [review]
LGTM

**Pull request merged**